### DISCUSSION TITLE
Screen margins

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#24] [Q&A] [answered]
It was requested in the past to offer an adjustable margin at the screen edges,
useful when putting our phones / tablets in cases that cover the screen edges.

Do we need a common margin for the 4 edges of the screen?
Or only for up and left / right? (avoiding too many settings)

<img src="https://user-images.githubusercontent.com/3484020/214856484-97341499-8dec-4b40-9e54-2d209189e613.jpg" width="50%"/>

**Comment by Ciutnik** [comment]
Interesting idea, although I'm not sure if it's needed for panels. I remember once there was a request to be able to offset only the buttons from the edges, and I see some usefulness here.

**Comment by Cer-o** [comment]
Only for up and left / right.

**Comment by DolomitesTom** [comment]
>a request to be able to offset only the buttons from the edges, and I see some usefulness here.

yes, it was my request and it was only related to the action buttons in order to be able to use them with motorcycle gloves, even more with thick winter gloves when the smartphone is in special cases. That would be awesome, in portrait and landscape mode.

**Comment by DolomitesTom** [comment]
An offset for the info bars is not necessary and visually not so pretty, only the buttons

**Comment by devemux86** [comment]
Don't cases usually hide the screen on all sides?
Are there examples of cases or pictures of phones in them to see what it looks like?

**Comment by DolomitesTom** [comment]
> Don't cases usually hide the screen on all sides?

It's not a matter of vision it's matter of usability. I use big font size and therefore the information in info bars are readable. Also the direction/arrow field is big enough to use with motorcycle gloves.

> Are there examples of cases or pictures of phones in them to see what it looks like?

I think i had uploaded a picture in the kurviger forum, have to search for it.

**Comment by DolomitesTom** [comment]
![20230126_181715](https://user-images.githubusercontent.com/121371629/214904467-b43a3abb-4c4d-41cc-a15d-ed7670692e2f.jpg)

![20230126_181726](https://user-images.githubusercontent.com/121371629/214904555-f4180c44-d9d1-4ef5-a504-f40e66434bdb.jpg)

Due to this lateral holders it is almost impossible to hit the buttons correctly with thick modecycle gloves. The spacing between the buttons should be increased the best and the offset from the left and right borders.

**Reply by devemux86** [reply]
Thanks for the images.

> The spacing between the buttons should be increased the best and the offset from the left and right borders.

Instead of increasing the font size, have you tried the "Settings | Application | Display size"?
It changes both fonts + user interface (sizes and spacing).

**Reply by DolomitesTom** [reply]
Yes with different settings, now at 110%. The higher I set it, the less information can be displayed at the right hand side. And it just makes the buttons a little bit bigger, and the action menu and gps button is still too much at the edge without offset. If the offset of the buttons cannot be set independently from other navigation bars, I am also happy with a common margin.

**Comment by devemux86** [comment]
See an image in OsmAnd for another case: https://github.com/osmandapp/OsmAnd/issues/14569.

**Comment by devemux86** [comment]
Now that we can have separate groups of settings (https://github.com/devemux86/cruiser/discussions/31), we can provide detailed margin settings.

Do we need for all 4 sides?
And should they only affect the bottom buttons?

**Comment by Ciutnik** [comment]
In my opinion, useful only for buttons (yes, on all four sides), but to please everyone, maybe a checkbox, e.g. 'Apply to panels also', would be needed?

**Reply by DolomitesTom** [reply]
My focus is on buttons, but the idea with "apply to panels additionally" is good

**Comment by devemux86** [comment]
> In my opinion, useful only for buttons (yes, on all four sides), but to please everyone, maybe a checkbox, e.g. 'Apply to panels also', would be needed?

> My focus is on buttons, but the idea with "apply to panels additionally" is good

<img src="https://user-images.githubusercontent.com/3484020/217525493-93d42f12-a37c-40ab-a11f-d8ae79eff7a9.jpg" width="50%"/><img src="https://user-images.githubusercontent.com/3484020/217525500-63997eb9-3b76-4392-981e-5160cf3f8b45.jpg" width="50%"/>

**Reply by DolomitesTom** [reply]
Meets the ideas exactly. The gloves are already looking forward 😁

**Reply by Ciutnik** [reply]
Perfectly! Nothing more, nothing less 👍

**Comment by devemux86** [comment, marked as answer]
Adjustable screen margin for screen panels and / or buttons is available in [Cruiser 3.0.9](https://play.google.com/store/apps/details?id=gr.talent.cruiser):

- Settings | Application | Appearance | Screen margin
- Or search settings for "margin"

**Reply by DolomitesTom** [reply]
Bearish, works perfect.

**Comment by DolomitesTom** [comment]
The display borders are a perfect new feature. My preferred settings are 45px left and right and 20px from the bottom. But in planning mode this is a little bit disturbing, especially when zooming the map. This causes an unwanted behavior. I would need the borders only in navigation mode for better handling with motorcycle gloves. Could you please offer an additional option to activate/deactivate borders for these individual modes?

**Reply by devemux86** [reply]
> Could you please offer an additional option to activate/deactivate borders for these individual modes?

Now that the settings are organized into separate screens, we can easily do this.
I have put it in my list. :slightly_smiling_face:

**Reply by devemux86** [reply]
> Could you please offer an additional option to activate/deactivate borders for these individual modes?

Implemented in [Cruiser 3.0.10](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by devemux86** [comment]
<img src="https://user-images.githubusercontent.com/3484020/219872749-0506bb2b-e4cb-44c7-82e9-81d4979dad97.jpg" width="50%"/>

**Reply by DolomitesTom** [reply]
Great, thanks a lot

**Comment by DolomitesTom** [comment]
this is soooo good, tested today on an afternoon tour... how is it realized? In the navigation mode, can the buttons also be set back (margins not apply) if the navigation is only paused? Because then it is more convenient to select a wp or add a wp and zoom and so on... when the navigation is then proceeded the margins should again apply...

**Reply by devemux86** [reply]
> this is soooo good, tested today on an afternoon tour...

:heart: 

> In the navigation mode, can the buttons also be set back (margins not apply) if the navigation is only paused?

I was wondering what we should do in pause mode, sure we can set the margin only in active navigation.
(will be available in the next Cruiser update)

**Reply by DolomitesTom** [reply]
Perfect thanks

**Reply by devemux86** [reply]
> In the navigation mode, can the buttons also be set back (margins not apply) if the navigation is only paused?

Implemented in [Cruiser 3.0.10](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by DolomitesTom** [reply]
Works, thanks a lot

**Comment by devemux86** [comment]
> the action buttons in order to be able to use them with motorcycle gloves, even more with thick winter gloves when the smartphone is in special cases

Next Cruiser will have larger buttons and spacing, easier to use outside with gloves, etc.

There will also be more color in the interface for a more pleasant user experience.

**Reply by DolomitesTom** [reply]
Cool

**Reply by MotoUKRider** [reply]
Is it sad that I've brought my summer gloves in from the garage (with the touchscreen fingertips) purely to test these features?!

**Comment by devemux86** [comment]
[Cruiser 3.0.11](https://play.google.com/store/apps/details?id=gr.talent.cruiser) has larger buttons and spacing, easier to use outside with gloves, etc.

Also a new accent color (blue) for better visibility. :blue_heart: 
It would be nice to let users choose the accent color, but Android does not support it.
